### PR TITLE
MHV-64687: Add a "Reset Filter" button

### PR DESCRIPTION
**Pull request by vmenshutin-bylight**
## Summary

Added a "Reset filter" button to `MedicationsListFilter` component

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-64687

## Testing done

- Manual testing
- Unit testing

## Screenshots

image
image

## What areas of the site does it impact?

/my-health/medications

## Acceptance criteria

AC1: Add "Reset filter" button to the filters that changes the list back to the "All medications" remote button option. 
AC2: Make sure this also changes the Showing.... as going back to all medications as well. 
AC3: Button will match the size of the apply filter button. 

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv-medications/components/MedicationsList/MedicationsListFilter.jsx
Comment (agravell047) [inline src/applications/mhv-medications/components/MedicationsList/MedicationsListFilter.jsx:131]: could we remove the extra div's here? It adds an unnecessary empty element to the accessibility tree:

Screenshot 2024-11-29 at 8 46 37 AM

Comment (vmenshutin-bylight) [reply src/applications/mhv-medications/components/MedicationsList/MedicationsListFilter.jsx:131]: I added div's for `display: block`. Any alternative suggestions please?

Comment (vmenshutin-bylight) [reply src/applications/mhv-medications/components/MedicationsList/MedicationsListFilter.jsx:131]: Unfortunately it won't work:

image

Comment (vmenshutin-bylight) [reply src/applications/mhv-medications/components/MedicationsList/MedicationsListFilter.jsx:131]: I would prefer to leave as is and apply display: block to a container rather then a button itself

Comment (agravell047) [reply src/applications/mhv-medications/components/MedicationsList/MedicationsListFilter.jsx:131]: Recap: the CSS for `filter-submit-btn `(our custom class) has width set to 100%. So I would use `vads-u-width--auto` and `vads-u-display--block` to get this instead of using the DIVs and using a custom class for width 100% unnecessarily. Also waiting on answer about which is the right breakpoint for this button because it seems to be turning into 100% width too early.

Approving to fix this and if changes are needed it will be done in a separate ticket as to next mess with staging review.

**Review by agravell047 — APPROVED** [review]

**Review by robertroddev — APPROVED** [review]

**Pull request merged**
